README.md

```diff
@@ -36,21 +45,21 @@ As CPUSetter is triggered by all Pods on all Nodes, we can be sure no containers
+If container has multiple processes and multiple exclusive CPUs are allocated or different pool types are used, pod annotation can be used to configure processes and CPU amounts for them.
+In this case CPU Pooler takes care of pinning the processes to allocated -but only to the allocated- CPUs. This is suitable for single threaded processes running on exclusive CPUs. The container can also have process(es) running on shared CPUs
 
 **use environment variables for pinning**
 
-CPU Pooler sets allocated exclusive CPUs to environment variable `EXCLUSIVE_CPUS` and allocated shared CPUs to `SHARED_CPUS` environment variable. They contain CPU(s) as comma separated list. These variables can be used by the application to read the allocated CPU(s) and do pinning of threads / processes to the CPU(s). This option needs to be used e.g for multithreaded processess with exlusive CPUs.
+CPU Pooler sets allocated exclusive CPUs to environment variable `EXCLUSIVE_CPUS` and allocated shared CPUs to `SHARED_CPUS` environment variable. They contain CPU(s) as comma separated list. These variables can be used by the application to read the allocated CPU(s) and do pinning of threads / processes to the CPU(s).
 
-In this option there is a synchronization problem in CPUsetter setting the cpuset and application doing the pinning (by setting the CPU affinity); if application does the pinning immediately after the startup, it is highly likely that CPUsetter has not finished it's job and cpuset is set after the CPU affinity by the application process. In this case the affinity setting is lost. This problem is solved by the `process-starter`, which waits for finishing of cpuset configuration and starts the container process only after that. The `process-starter` can be used only if the `command` property is configured in container's pod manifest. If the `command` is not configured, the `process-starer` does not know which process to start in the container. If `command` cannot be configured, application must make sure that cpuset is configured (=matches to union of SHARED_CPUS and EXCLUSIVE_CPUS) before setting the CPU affinity.
+To avoid possible race conditions occuring due to multiple process trying to set affinity at the same time, or application trying to set it too early; CPU-Pooler guarantees that the container's entrypoint is only executed once the proper CPU configuration has been provisioned.
+This is achieved by the `process-starter` component. Please note, that in order for CPU-Pooler to work as intended the `command` property must be configured in container's pod manifest. If the `command` is not configured, the `process-starer` does not know which process to start in the container, and exotic race conditions might occur.
```

Comment:
If the `command` is not given, the `process-starter` is not used at all. Maybe that could be described here as now it sounds like the `process-starter` is used but it does not know what to do.